### PR TITLE
Mobile ucr filters

### PR DESCRIPTION
**Pull request by nickpell**
For review: @czue cc: @gcapalbo 

The UX is pretty painful right now (over-exposes the underlying models), so I want to fix that to be in line with the suggested UX in the spec.

Note that this contains commits from https://github.com/dimagi/commcare-hq/pull/7698

Ideally I would first like to merge https://github.com/dimagi/commcare-hq/pull/7698 and https://github.com/dimagi/commcare-hq/pull/7683 (still holding off on merge for the latter) and then handle the resulting merge conflicts, then merge to master.

screen shot 2015-07-30 at 5 42 11 pm

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 13 errors:

* corehq/apps/app_manager/models.py, line 14 - F401 'date' imported but unused
* corehq/apps/app_manager/views.py, line 925 - E261 at least two spaces before inline comment
* corehq/apps/app_manager/views.py, line 926 - E261 at least two spaces before inline comment
* corehq/apps/app_manager/static/app_manager/js/report-module.js, line 28 - Don't make functions within a loop.
* corehq/apps/app_manager/static/app_manager/js/report-module.js, line 30 - Use '!==' to compare with 'undefined'.
* corehq/apps/app_manager/static/app_manager/js/report-module.js, line 47 - ['doc_type'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/report-module.js, line 50 - ['filter_type'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/report-module.js, line 52 - ['date_range'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/report-module.js, line 54 - ['custom_data_property'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/report-module.js, line 56 - ['value'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/report-module.js, line 58 - ['select_value'] is better written in dot notation.
* corehq/apps/app_manager/static/app_manager/js/report-module.js, line 78 - Missing semicolon.
* corehq/apps/app_manager/static/app_manager/js/report-module.js, line 114 - 'i' is already defined.

File: corehq/apps/app_manager/static/app_manager/js/report-module.js [hunk 1]
```diff
@@ -1,17 +1,98 @@
...
+                }
+                filter.selectedValue.filter_type = ko.observable(filter.selectedValue.filter_type || '');
+                filter.selectedValue.date_range = ko.observable(filter.selectedValue.date_range || '');
+                filter.selectedValue.custom_data_property = ko.observable(filter.selectedValue.custom_data_property || '');
+                filter.selectedValue.value = ko.observable(filter.selectedValue.value ? filter.selectedValue.value.join("\u001F") : '');
+                filter.selectedValue.select_value = ko.observable(filter.selectedValue.select_value || '');
+
+                filter.dynamicFilterName = ko.computed(function() {
+                    return reportId() + '/' + filter.slug;
+                });
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/report-module.js]: Don't make functions within a loop.

File: corehq/apps/app_manager/static/app_manager/js/report-module.js [hunk 2]
```diff
@@ -1,17 +1,103 @@
...
+                    'select_value'
+                ];
+                for(var filterFieldsIndex in filterFields) {
+                    filter.selectedValue[filterFields[filterFieldsIndex]] = ko.observable(filter.selectedValue[filterFields[filterFieldsIndex]] || '');
+                }
+                filter.selectedValue.value = ko.observable(filter.selectedValue.value ? filter.selectedValue.value.join("\u001F") : '');
+
+                filter.dynamicFilterName = ko.computed(function() {
+                    return reportId() + '/' + filter.slug;
+                });
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/report-module.js]: Don't make functions within a loop.

**Comment by czue** [comment]
fwiw I think the UX is probably fine at this stage (we're in heavy R&D mode on this feature currently)

File: corehq/apps/app_manager/templates/app_manager/module_view_report.html [hunk 3]
```diff
@@ -1,24 +1,74 @@
...
+            var initialValues = element.val() !== "" ? element.val().split(separator) : [];
+            element.select2({
+                minimumInputLength: 0,
+                multiple: true,
+                separator: separator,
+                allowClear: true,
+                // allowClear only respected if there is a non empty placeholder
+                placeholder: " ",
+                ajax: {
+                    url: "/a/staging/configurable_reports/api/choice_list/" + element.parent()[0].lastElementChild.value + "/",
```
Comment (czue) [inline corehq/apps/app_manager/templates/app_manager/module_view_report.html]: this should use `{% url %}` and definitely shouldn't hard code the domain :)

File: corehq/apps/app_manager/views.py [hunk 4]
```diff
@@ -1867,6 +1868,7 @@ def edit_report_module(request, domain, app_id, module_id):
     module = app.get_module(module_id)
     assert isinstance(module, ReportModule)
     module.name = params['name']
+    print params['reports']
```
Comment (czue) [inline corehq/apps/app_manager/views.py]: print

**Comment by czue** [comment]
looking good overall! i didn't review the front-end code super closely

File: corehq/apps/app_manager/templates/app_manager/module_view_report.html (hunk 3 above)
Comment (nickpell) [inline corehq/apps/app_manager/templates/app_manager/module_view_report.html]: haha valid point

**Comment by nickpell** [comment]
closing in favor of https://github.com/dimagi/commcare-hq/pull/7760

**Pull request closed without merging**